### PR TITLE
planner: fix explain panic when using VALUES statement

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/45957

Problem Summary:

### What is changed and how it works?

Fixed the explain panic issue when using the VALUES statement.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fixed the Explain panic problem when using the VALUES statement
```

### CODE REVIEW

File: planner/core/plan_test.go [hunk 1]
```diff
@@ -754,3 +754,12 @@ func TestIssue40535(t *testing.T) {
...
 }
+
+func TestExplainValuesStatement(t *testing.T) {
+	store, _ := testkit.CreateMockStoreAndDomain(t)
+	tk := testkit.NewTestKit(t, store)
+
+	_, err := tk.Exec("EXPLAIN FORMAT = TRADITIONAL ((VALUES ROW ()) ORDER BY 1)")
+	require.Error(t, err)
+	require.Regexp(t, ".*Unknown table ''.*", err.Error())
+}
```
Comment (hawkingrei) [inline planner/core/plan_test.go:765]: you can use ```MustMatchErrMsg```.

Comment (hi-rustin) [reply planner/core/plan_test.go]: Good suggestion! Thanks!

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**